src/block.js

```diff
@@ -133,6 +136,43 @@ Block.calculateTarget = function (bits) {
   return target
 }
 
+Block.calculateMerkleRoot = function (transactions) {
+  if (transactions.length === 0) return NULL_HASH
+
+  var hashes = transactions.map(function (transaction) {
+    return transaction.getHash()
+  })
+
+  var i = transactions.length
+  var k = 0
+
+  while (i > 1) {
+    for (var j = 0; j < i; j += 2) {
+      var m = Math.min(i - 1, j + 1)
+      var hash1 = hashes[j + k]
+      var hash2 = hashes[k + m]
+
+      var hasher = createHash('sha256')
+      hasher.update(hash1)
+      hasher.update(hash2)
+
+      hashes.push(bcrypto.sha256(hasher.digest()))
+    }
+
+    k += i
+    i = (i + 1) >> 1
+  }
+
+  return hashes[hashes.length - 1]
```

Comment:
Sorry, I not tested snippet, just concept. I think this wrong: `; j < length;`, `i` should be here.